api/api.py

```diff
@@ -509,10 +505,22 @@ def bids(user, key):
 
 @api_blueprint.route('/api/users/<string:nym>/auctions', methods=['GET'])
 def user_auctions(nym):
+    token = get_token_from_request()
+    for_user_id = None
+    if token:
+        for_user_id = get_user_from_token(token).id
     user = m.User.query.filter_by(nym=nym).first()
     if not user:
         return jsonify({'message': "No auctions found"}), 404
-    return jsonify({'auctions': [a.to_dict() for a in sorted(user.auctions.all(), key=lambda a: a.created_at, reverse=True)]})
+    now = datetime.now()
+    auction_status = request.args.get('filter')
```

Comment:
What if no filter is passed? It seems like we assume it is optional (by using .get) and yet we don't treat that case. I think from an API design perspective it would be nice to keep it optional although not needed by our web app - and just return all auctions in that case.